dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -207,6 +209,8 @@ public static void main(String[] args) throws Exception {
                         .withProgressModel(o.getProgressModel())
                         .withWitness(witness);
                 Program p = new ProgramParser().parse(f);
+                p.setEntrypoint(new Entrypoint.Simple(p.getFunctionByName(o.getEntryFunction()).orElseThrow(
+                    () -> new MalformedProgramException(String.format("Program has no function named %s. Select a different entry point.", o.getEntryFunction())))));
```

Comment:
Isn't this piece of code always executed? Wouldn't it fail on litmus or SPIRV code that does not have a "main"?
I think you might want to put as default value for the entrypoint the empty string with the meaning that it defaults to whatever the parser decides. Only for non-empty values you overwrite the parser's decision.